sympy/functions/elementary/tests/test_trigonometric.py

```diff
@@ -923,6 +922,13 @@ def test_asin_series():
     assert t5 == 3*x**5/40
     assert asin(x).taylor_term(7, x, t5, 0) == 5*x**7/112
 
+def test_asin_leading_term():
+    assert asin(x).as_leading_term(x) == x
+    assert asin(x + 1).as_leading_term(x) == pi/2
+    assert asin(x - 1).as_leading_term(x) == -pi/2
+    assert asin(1/x).as_leading_term(x, cdir=1) == I*log(x) + pi/2 - I*log(2)
+    assert asin(1/x).as_leading_term(x, cdir=-1) == -I*log(x) - 3*pi/2 + I*log(2)
```

Comment:
It looks like there should be `-pi/2` instead of `-3*pi/2`:
```
In [49]: asin(-2).n()
Out[49]: -1.5707963267949 + 1.31695789692482⋅ⅈ
In [50]: asin(-10).n()
Out[50]: -1.5707963267949 + 2.99322284612638⋅ⅈ
In [51]: asin(-100).n()         
Out[51]: -1.5707963267949 + 5.29829236561048⋅ⅈ
```
It is probably best to comment this out now and fix in a later PR. Something like this can also be expected in case of other inverse functions.

Reply:
I guess it is correct , atleast that's what the graph and wolfram shows ( I hope you didn't forget the negative cdir here ) 
![image](https://user-images.githubusercontent.com/87052487/176861725-ea4be5de-d81d-49a8-a402-3a7426ba37d0.png)

Also we can achieve this on master through gruntz 
```
>>> x = Symbol('x')
>>> asin(1/x).rewrite(log)
-I*log(sqrt(1 - 1/x**2) + I/x)
>>> expand_log(_, force=True)
-I*log(sqrt(1 - 1/x**2) + I/x)
>>> gruntz(-I*log(sqrt(1 - 1/x**2) + I/x) + I*log(x), x, 0, '-')
I*(log(2) + 3*I*pi/2)
```

Reply:
The difference between `-pi/2` and `-3*pi/2` is accounted for by the imaginary part of `log(x)` for negative `x`. So there is no problem after all..